third_party/move/move-compiler-v2/tests/checking/specs/expressions_err.exp

```diff
@@ -12,37 +12,37 @@ error: no function named `not_declared` found
 15 │       not_declared() // Undeclared function.
    │       ^^^^^^^^^^^^^^
 
-error: expected `num` but found `bool`
+error: cannot return `bool` from a function with result type `num`
    ┌─ tests/checking/specs/expressions_err.move:19:7
    │
 19 │       false // Wrong result type.
    │       ^^^^^
 
-error: invalid call of `>`: expected `num` but found `vector<num>` for argument 1
+error: cannot use `vector<num>` with an operator which expects a value of type `num`
    ┌─ tests/checking/specs/expressions_err.move:23:7
    │
 23 │       x > y // No matching function.
    │       ^
 
-error: expected `(num, bool)` but found `bool`
+error: cannot return `bool` from a function with result type `(num, bool)`
    ┌─ tests/checking/specs/expressions_err.move:27:7
    │
 27 │       false // Wrong result type tuple.
    │       ^^^^^
 
-error: invalid call of `M::wrongly_typed_callee`: expected `bool` but found `u256` for argument 2
+error: cannot use `u256` with an operator which expects a value of type `bool`
```

Comment:
Using "operator" here is awkward.  Perhaps we can distinguish `Operation::MoveFunction` from other operations?  Also, we've lost the argument number.

Reply:
Haven't you seen that its distinguishing this all over the place already? Only spec functions where forgotten.